src/managed/OpenLiveWriter/ApplicationMain.cs

```diff
@@ -190,6 +196,90 @@ private static bool LaunchAction(string[] args, bool isFirstInstance)
             }
         }
 
+        /// <summary>
+        /// Registers functions to handle Squirrel's events.
+        /// </summary>
+        /// <param name="downloadUrl">The Url to use for downloading payloads.</param>
+        private static void RegisterSquirrelEventHandlers(string downloadUrl)
+        {
+            try
+            {
+                using (var mgr = new Squirrel.UpdateManager(downloadUrl))
+                {
+                    SquirrelAwareApp.HandleEvents(
+                        onInitialInstall: v => InitialInstall(mgr),
+                        onFirstRun: () => FirstRun(mgr),
+                        onAppUpdate: v => OnAppUpdate(mgr),
+                        onAppUninstall: v => OnAppUninstall(mgr));
+                }
+            }
+            catch (Exception)
+            {
+                // Ignore, chances are this is being run locally on a dev's machine
+            }
```

Comment:
Use a DEBUG tag to support dev box. If this fails in release code the error should be logged.